### PR TITLE
Merge 1.93 from upstream

### PR DESCRIPTION
**Pull request by jmcphers**
This change includes two large payloads:

- All the changes from Code OSS 1.92.0 -> 1.93.0
- All of the Posit Workbench modifications from their fork of VS Code

A lot of hand-editing needed to happen here since our compressed history means we have no common baseline to use for a commit-based merge. The methodology used was roughly:

1. Merge 1.93.0 into vscode-server and test against Workbench; make fixes/tweaks as needed
2. Merge 1.93.0 into Positron (using our usual merge strategy)
3. Copy all the code files (which now have a common 1.93.0 baseline) from vscode-server => Positron
4. Go through the git view for each one and reconcile the Positron and PWB change hunks, discarding the edits that dropped Positron blocks, staging the edits that added PWB blocks, and manually resolving the dozen or so places where we both edited the same logic
5. Commit and continue typical upstream merge process

This was a one-shot merge process that won't need to be repeated since next time we will have a common baseline (a tag from vscode-server representing what we merged last time).

Nothing too crazy in here. One of the more annoying parts was getting the `rstudio.rstudio-workbench` built-in extension set up so that it gets bundled for the `reh-web` build but ignored in other contexts. 

### QA Notes

This contains a lot of churn introduced from two different sources, so some testing is indicated pre-merge. Here are some binaries you can test:

https://github.com/posit-dev/positron-builds/actions/runs/10821973170

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/views/openEditorsView.ts [hunk 1]
```diff
@@ -902,23 +897,6 @@ registerAction2(class extends Action2 {
...
-		});
-	}
-
-	async run(accessor: ServicesAccessor): Promise<void> {
-		const commandService = accessor.get(ICommandService);
-		await commandService.executeCommand(SAVE_ALL_TITLED_COMMAND_ID);
-	}
-});
-// --- End Positron ---
-
```
Comment (wesm) [inline src/vs/workbench/contrib/files/browser/views/openEditorsView.ts:921]: Are these deletions intentional?

**Comment by jmcphers** [comment]
This change introduced a surprising number of test failures that I still need to work through before merge. Right now the positron-r tests are failing because the test process is literally exiting in the middle of the test for no apparent reason.

```
[main 2024-09-12T21:00:20.713Z] [UtilityProcessWorker]: terminated unexpectedly with code 1863645568, signal: unknown
```

File: remote/web/package.json [hunk 2]
```diff
@@ -6,25 +6,25 @@
...
-		"@xterm/addon-webgl": "0.19.0-beta.51",
+		"@xterm/addon-clipboard": "0.2.0-beta.35",
+		"@xterm/addon-image": "0.9.0-beta.52",
+		"@xterm/addon-search": "0.16.0-beta.52",
+		"@xterm/addon-serialize": "0.14.0-beta.52",
+		"@xterm/addon-unicode11": "0.9.0-beta.52",
+		"@xterm/addon-webgl": "0.19.0-beta.52",
+		"@xterm/xterm": "5.6.0-beta.52",
 		"@xterm/xterm": "5.6.0-beta.51",
-		"he": "^1.2.0",
```
Comment (sharon-wang) [inline remote/web/package.json:17]: We lost `he`! Can we please bring `he` back? We'll need to `yarn` so the reh-web `package.json` can re-inherit `he` as well

Comment (jmcphers) [reply remote/web/package.json:17]: done!

File: remote/web/package.json [hunk 3]
```diff
@@ -6,25 +6,26 @@
...
-		"@xterm/addon-unicode11": "0.9.0-beta.51",
-		"@xterm/addon-webgl": "0.19.0-beta.51",
+		"@xterm/addon-clipboard": "0.2.0-beta.35",
+		"@xterm/addon-image": "0.9.0-beta.52",
+		"@xterm/addon-search": "0.16.0-beta.52",
+		"@xterm/addon-serialize": "0.14.0-beta.52",
+		"@xterm/addon-unicode11": "0.9.0-beta.52",
+		"@xterm/addon-webgl": "0.19.0-beta.52",
+		"@xterm/xterm": "5.6.0-beta.52",
 		"@xterm/xterm": "5.6.0-beta.51",
```
Comment (sharon-wang) [inline remote/web/package.json:18]: Just noticed we have a duplicate entry here for `"@xterm/xterm"`. Based on `remote/package.json`, I think we want the `.52` one!

File: remote/web/package.json [hunk 4]
```diff
@@ -6,25 +6,26 @@
...
-		"he": "^1.2.0",
+    "he": "^1.2.0",
```
Comment (sharon-wang) [inline remote/web/package.json:19]: looks like the indentation changed from tabs to spaces on this line
```suggestion
		"he": "^1.2.0",
```

File: remote/web/package.json (hunk 3 above)
Comment (jmcphers) [reply remote/web/package.json]: fixed in https://github.com/posit-dev/positron/pull/4655/commits/0ebafbf76c1da36368b1ca2071c8f5058e1d3086!

File: remote/web/package.json (hunk 4 above)
Comment (jmcphers) [reply remote/web/package.json]: fixed in https://github.com/posit-dev/positron/pull/4655/commits/0ebafbf76c1da36368b1ca2071c8f5058e1d3086!

**Review by sharon-wang — DISMISSED** [review]
LGTM!

I've been running a build based on this branch in Workbench and in Server+Web and Desktop locally with no obvious issues with the typical IDE functionality

**Pull request merged**